### PR TITLE
fix(compat): update patch for `typescript@4.6.0-beta`

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

The PnP compatibility patch for TypeScript doesn't apply to `typescript@>=4.6.0-beta`.

**How did you fix it?**

Rebased https://github.com/microsoft/TypeScript/pull/35206 onto the `4.6.0-beta` release

**Checklist**
- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).
- [x] I have set the packages that need to be released for my changes to be effective.
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/plugin-compat/extra/typescript/gen-typescript-patch.js [hunk 1]
```diff
@@ -80,10 +80,17 @@ const SLICES = [
   },
   // https://github.com/merceyz/TypeScript/tree/merceyz/pnp-4.5
   {
-    from: `79fe5d206f89e5f1585807bdcd7feeb567a69646`,
-    to: `fff89a12766fa485e35b1c65d8631d109c4e2c0e`,
-    onto: `7893f689fedcfe28a499d119d34844840bce9160`,
-    range: `>=4.5.2`,
+    from: `9232978f8e54f073b5451d0bf2737d42a0fd672f`,
+    to: `3a2388d39d41d000b5c5f9bcd48096b39fcedf8f`,
+    onto: `55e13e9115b3cc5458d76c39da1211dc28d7b51f`,
```
Comment (arcanis) [inline packages/plugin-compat/extra/typescript/gen-typescript-patch.js:85]: Why did you need to change the from/to/onto?

Comment (merceyz) [reply packages/plugin-compat/extra/typescript/gen-typescript-patch.js:85]: There has been a few releases for 4.5 since we last generated the patch so I updated my branch, I'll revert it since it isn't strictly needed.
